Vulkan-Pipeline-Assistant/memoryallocator.cpp

```diff
@@ -5,9 +5,42 @@
 
 using namespace vpa;
 
-MemoryAllocator::MemoryAllocator(QVulkanDeviceFunctions* deviceFuncs, QVulkanWindow* window) : m_deviceFuncs(deviceFuncs), m_window(window) { }
+MemoryAllocator::MemoryAllocator(QVulkanDeviceFunctions* deviceFuncs, QVulkanWindow* window) : m_deviceFuncs(deviceFuncs), m_window(window) {
+    VkQueueFamilyProperties queueProperties;
+    QVulkanFunctions* funcs = window->vulkanInstance()->functions();
+    uint32_t queueCount = 0;
+    funcs->vkGetPhysicalDeviceQueueFamilyProperties(window->physicalDevice(), &queueCount, nullptr);
+    QVector<VkQueueFamilyProperties> queueFamilies(queueCount);
+    funcs->vkGetPhysicalDeviceQueueFamilyProperties(window->physicalDevice(), &queueCount, queueFamilies.data());
+    for (uint32_t i = 0; i < queueFamilies.size(); ++i) {
+        if (queueFamilies[i].queueFlags & VK_QUEUE_TRANSFER_BIT) {
+            m_transferQueueIdx = i;
+            break;
+        }
+    }
+    deviceFuncs->vkGetDeviceQueue(window->device(), m_transferQueueIdx, 0, &m_transferQueue);
+
+    VkCommandPoolCreateInfo poolInfo = {};
+    poolInfo.sType = VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO;
+    poolInfo.queueFamilyIndex = m_transferQueueIdx;
+    poolInfo.flags = 0;
+
+    VkResult result = m_deviceFuncs->vkCreateCommandPool(window->device(), &poolInfo, nullptr, &m_commandPool);
+    if (result != VK_SUCCESS) qFatal("Failed to allocate command pool for memory allocator!");
 
-MemoryAllocator::~MemoryAllocator() { }
+    VkCommandBufferAllocateInfo allocInfo = {};//m_commandBuffer
+    allocInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+    allocInfo.commandPool = m_commandPool;
+    allocInfo.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
+    allocInfo.commandBufferCount = 1;
+
+    result = m_deviceFuncs->vkAllocateCommandBuffers(window->device(), &allocInfo, &m_commandBuffer);
+    if (result != VK_SUCCESS) qFatal("Failed to allocate command buffer for transfer!");
```

Comment:
Checking if vulkan functions returning VK_SUCCESS should be macro'd for good coding practices.

Reply:
Good point, I'll make a note on trello to do this when refactoring to have a proper error handling system.